src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -187,10 +193,13 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     {
                         return;
                     }
+                    
+                    IAsyncEnumerator<ConfigurationClient> clientEnumerator = _configClientManager
+                        .GetAvailableClients(cancellationToken)
+                        .GetAsyncEnumerator();
 
-                    IEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients(utcNow);
-
-                    if (!availableClients.Any())
+                    await using ConfiguredAsyncDisposable configuredAsyncDisposable = clientEnumerator.ConfigureAwait(false);
```

Comment:
what is this used for?

Reply:
For registering the enumerator for dispose eh?

Reply:
Yes, because if call this directly, it's not allowed since `ConfigAwait(false)` will cause type mismatch.
It has to be `ConfiguredAsyncDisposable` type, and don't worry, dispose `ConfiguredAsyncDisposable` will also dispose the origin `IAsyncEnumerator`.
`await using IAsyncEnumerator<ConfigurationClient> clientEnumerator =_configClientManager
                        .GetAvailableClients(cancellationToken)
                        .GetAsyncEnumerator().ConfigureAwait(false); `